test/SILOptimizer/move_only_checker_trivial.swift

```diff
@@ -0,0 +1,410 @@
+    @_noImplicitCopy let x2 = x // expected-error {{'x2' consumed more than once}}
+    print(x2.lhs) // expected-note {{consuming use}}
+    for _ in 0..<1024 {
+        print(x2.lhs) // expected-note {{consuming use}}
+    }
+}
+
+////////////////////////////////////////////////////////////
+// Aggregate Generic Struct + Generic But Body is Trivial //
+////////////////////////////////////////////////////////////
+
+public func aggGenericStructUseMoveOnlyWithoutEscaping<T>(_ x: AggGenericStruct<T>) {
+}
+public func aggGenericStructConsume<T>(_ x: __owned AggGenericStruct<T>) {
+}
+
+public func aggGenericStructSimpleChainTest<T>(_ x: AggGenericStruct<T>) {
+    @_noImplicitCopy let x2 = x
+    let y2 = x2
+    let k2 = y2
+    aggGenericStructUseMoveOnlyWithoutEscaping(k2)
+}
+
+public func aggGenericStructSimpleNonConsumingUseTest<T>(_ x: AggGenericStruct<T>) {
+    @_noImplicitCopy let x2 = x
+    aggGenericStructUseMoveOnlyWithoutEscaping(x2)
+}
+
+public func aggGenericStructMultipleNonConsumingUseTest<T>(_ x: AggGenericStruct<T>) {
+    @_noImplicitCopy let x2 = x
+    aggGenericStructUseMoveOnlyWithoutEscaping(x2)
+    aggGenericStructUseMoveOnlyWithoutEscaping(x2)
+    print(x2)
+}
+
+public func aggGenericStructUseAfterConsume<T>(_ x: AggGenericStruct<T>) {
+    @_noImplicitCopy let x2 = x // expected-error {{'x2' consumed more than once}}
+    aggGenericStructUseMoveOnlyWithoutEscaping(x2)
+    let y = x2 // expected-note {{consuming use}}
+    let _ = y
+    aggGenericStructConsume(x2)
+    print(x2) // expected-note {{consuming use}}
+}
+
+public func aggGenericStructDoubleConsume<T>(_ x: AggGenericStruct<T>) {
+    @_noImplicitCopy let x2 = x  // expected-error {{'x2' consumed more than once}}
+    let y = x2 // expected-note {{consuming use}}
+    let _ = y
+    let z = x2 // expected-note {{consuming use}}
+    let _ = z
+}
+
+public func aggGenericStructLoopConsume<T>(_ x: AggGenericStruct<T>) {
+    @_noImplicitCopy let x2 = x // expected-error {{'x2' consumed more than once}}
+    for _ in 0..<1024 {
+        let z = x2 // expected-note {{consuming use}}
+        let _ = z
+    }
+}
+
+public func aggGenericStructDiamond<T>(_ x: AggGenericStruct<T>) {
+    @_noImplicitCopy let x2 = x
+    if boolValue {
+        let z = x2
+        let _ = z
+    } else {
+        let z = x2
+        let _ = z
+    }
+}
+
+public func aggGenericStructDiamondInLoop<T>(_ x: AggGenericStruct<T>) {
+    @_noImplicitCopy let x2 = x // expected-error {{'x2' consumed more than once}}
+    for _ in 0..<1024 {
+        if boolValue {
+            let z = x2 // expected-note {{consuming use}}
+            let _ = z
+        } else {
+            let y = x2 // expected-note {{consuming use}}
+            let _ = y
+        }
+    }
+}
+
+public func aggGenericStructAccessField<T>(_ x: AggGenericStruct<T>) {
+    @_noImplicitCopy let x2 = x // expected-error {{'x2' consumed more than once}}
+    print(x2.lhs) // expected-note {{consuming use}}
```

Comment:
Projecting a stored property, or more generally, calling a `get` or `read` accessor with a guaranteed base, should not be a consuming use.

Reply:
The reason this is a consuming use is b/c of print. Print consumes values unfortunately due to the way we construct the array.